openmdao/jacobians/dictionary_jacobian.py

```diff
@@ -357,7 +358,7 @@ def set_col(self, system, icol, column):
                     arr[row_inds] = 0.  # zero out the rows that are covered by sparsity
                     nzs = np.nonzero(arr)[0]
                     if nzs.size > 0:
-                        self._errors.append(f"{system.msginfo}: User specified sparsity (rows/cols)"
-                                            f" for subjac '{of}' wrt '{wrt}' is incorrect. There "
-                                            f"are non-covered nonzeros in column {loc_idx} at "
-                                            f"row(s) {nzs}.")
+                        issue_warning(f"{system.msginfo}: User specified sparsity (rows/cols)"
```

Comment:
I thought about this a little more after our initial discussion, I think the right way to handle this would be similar to the way we handle inconsistent keys, where it's really an error and we just add it to the data structure so that it can be part of the check_partials report at the end.  I understand why you don't want it to bail out too early, but having an incorrect sparsity really is a significant error, and making it a warning could easily result it it being ignored, leading to wrong answers.

Reply:
I agree that we should be displaying the information in the output. Maybe along the lines of `Nonzero rows/cols in Jfd not in Jfwd/Jrev`. The text output of the current warning makes it really difficult to visualize where the problem lies. I'll work on this to add it to the check partials data for display.